### PR TITLE
Get rid of Python2 numeric relics

### PR DESCRIPTION
**Pull request by eumiro**
Migration from Python2 to Python3 changes some numeric behavior. In Python3:

- `round()`, `math.ceil()` and `math.floor()` return `int` instead of `float`
- true division `int / int` returns `float` and therefore it is not necessary to convert one of the parts to `float` or to multiply the expression by `1.0`

The code has been simplified. No functionality has been changed.

### CODE REVIEW

File: airflow/providers/celery/executors/celery_executor_utils.py [hunk 1]
```diff
@@ -298,7 +298,7 @@ def _get_many_using_multiprocessing(self, async_results) -> Mapping[str, EventBu
         num_process = min(len(async_results), self._sync_parallelism)
 
         with ProcessPoolExecutor(max_workers=num_process) as sync_pool:
-            chunksize = max(1, math.floor(math.ceil(1.0 * len(async_results) / self._sync_parallelism)))
+            chunksize = max(1, math.ceil(len(async_results) / self._sync_parallelism))
```
Comment (uranusjr) [inline airflow/providers/celery/executors/celery_executor_utils.py:301]: I wonder why the outer `floor` existed in the first place. That wouldn’t do anything on Python 2 either…

File: tests/test_utils/perf/perf_kit/repeat_and_time.py [hunk 2]
```diff
@@ -120,10 +119,10 @@ def monte_carlo(total=10000):
         for _ in range(0, total):
             x_val = random.random() ** 2
             y_val = random.random() ** 2
-            if math.sqrt(x_val + y_val) < 1.0:
+            if x_val + y_val < 1:
```
Comment (uranusjr) [inline tests/test_utils/perf/perf_kit/repeat_and_time.py]: This doesn’t look right?

Comment (eumiro) [reply tests/test_utils/perf/perf_kit/repeat_and_time.py]: Mathematically if `math.sqrt(x) < 1` then `x < 1` as well, but this is probably useless optimization that hides the original idea. So I have reverted this change (and only replaced `1.0` with `1`).

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Review by pierrejeambrun — APPROVED** [review]

File: airflow/providers/celery/executors/celery_executor_utils.py (hunk 1 above)
Comment (sethlm2) [reply airflow/providers/celery/executors/celery_executor_utils.py:301]: Exactly

**Pull request merged**